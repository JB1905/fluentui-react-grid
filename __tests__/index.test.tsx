import React from 'react';
import { render } from '@testing-library/react';

import CompoundGrid, { Grid, Row, Col } from '../src';

describe('Grid', () => {
  it.each`
    dir
    ${'ltr'}
    ${'rtl'}
    ${undefined}
  `('should render the grid with dir: $dir', ({ dir }) => {
    const { container } = render(
      <Grid dir={dir}>
        <Row>
          <Col>Hello</Col>
          <Col>World</Col>
        </Row>
      </Grid>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render the grid with custom params', () => {
    const { container } = render(
      <CompoundGrid>
        <CompoundGrid.Row>
          <CompoundGrid.Col>Hello</CompoundGrid.Col>

          <CompoundGrid.Col
            className="custom-class-name"
            sizeSm={12}
            sizeMd={8}
            sizeLg={6}
            sizeXl={4}
            sizeXxl={3}
            sizeXxxl={2}
            hiddenSm
            hiddenMd
            hiddenLg
            hiddenXl
            hiddenXxl
            hiddenXxxl
            smPush={1}
            smPull={2}
            mdPush={3}
            mdPull={4}
            lgPush={5}
            lgPull={6}
            xlPush={7}
            xlPull={8}
            xxlPush={9}
            xxlPull={10}
            xxxlPush={11}
            xxxlPull={12}
            hiddenMdDown
            hiddenMdUp
            hiddenLgDown
            hiddenLgUp
            hiddenXlDown
            hiddenXlUp
            hiddenXxlDown
            hiddenXxlUp
          >
            World
          </CompoundGrid.Col>
        </CompoundGrid.Row>
      </CompoundGrid>
    );

    expect(container).toMatchSnapshot();
  });

  it.todo('empty grid')
  it.todo('only col')
  it.todo('only row')
  it.todo('incorrect nesting')
});
