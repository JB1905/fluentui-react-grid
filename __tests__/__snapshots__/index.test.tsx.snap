// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should render grid with custom params 1`] = `
<div>
  <div
    class="ms-Grid"
    dir="ltr"
  >
    <div
      class="ms-Grid-row"
    >
      <div
        class="ms-Grid-col"
      >
        Hello
      </div>
      <div
        class="ms-Grid-col ms-sm12 ms-md8 ms-lg6 ms-xl4 ms-xxl3 ms-xxl2 ms-sm1 ms-md3 ms-lg5 ms-xl7 ms-xxl9 ms-xxl11 ms-sm2 ms-md4 ms-lg6 ms-xl8 ms-xxl10 ms-xxl12 ms-hiddenSm ms-hiddenMd ms-hiddenMdDown ms-hiddenMdUp ms-hiddenLg ms-hiddenLgDown ms-hiddenLgUp ms-hiddenXl ms-hiddenXlDown ms-hiddenXlUp ms-hiddenXxl ms-hiddenXxlDown ms-hiddenXxlUp ms-hiddenXxxl custom-class-name"
      >
        World
      </div>
    </div>
  </div>
</div>
`;

exports[`Grid should render grid with dir: ltr 1`] = `
<div>
  <div
    class="ms-Grid"
    dir="ltr"
  >
    <div
      class="ms-Grid-row"
    >
      <div
        class="ms-Grid-col"
      >
        Hello
      </div>
      <div
        class="ms-Grid-col"
      >
        World
      </div>
    </div>
  </div>
</div>
`;

exports[`Grid should render grid with dir: rtl 1`] = `
<div>
  <div
    class="ms-Grid"
    dir="rtl"
  >
    <div
      class="ms-Grid-row"
    >
      <div
        class="ms-Grid-col"
      >
        Hello
      </div>
      <div
        class="ms-Grid-col"
      >
        World
      </div>
    </div>
  </div>
</div>
`;

exports[`Grid should render grid with dir: undefined 1`] = `
<div>
  <div
    class="ms-Grid"
    dir="ltr"
  >
    <div
      class="ms-Grid-row"
    >
      <div
        class="ms-Grid-col"
      >
        Hello
      </div>
      <div
        class="ms-Grid-col"
      >
        World
      </div>
    </div>
  </div>
</div>
`;
